src/components/Grunnlag/sections/Pensjonsavtaler.tsx

```diff
@@ -1,97 +1,149 @@
 import React from 'react'
+import { useNavigate } from 'react-router-dom'
 
+import { ExclamationmarkTriangleFillIcon } from '@navikt/aksel-icons'
 import { Accordion, BodyLong, BodyShort, Link } from '@navikt/ds-react'
 import clsx from 'clsx'
 
+import { useAppSelector } from '@/state/hooks'
+import { selectSamtykke } from '@/state/userInput/selectors'
 import { formatAsDecimal } from '@/utils/currency'
 import { capitalize } from '@/utils/string'
 
 import { SectionContent } from './components/SectionContent'
 import { SectionHeader } from './components/SectionHeader'
-import { groupPensjonsavtalerByType } from './Pensjonsavtaler-utils'
+import {
+  groupPensjonsavtalerByType,
+  getPensjonsavtalerTittel,
+  getMaanedString,
+} from './utils'
 
 import styles from './Pensjonsavtaler.module.scss'
 
 interface Props {
   pensjonsavtaler: Pensjonsavtale[]
+  showError: boolean
 }
 
-export function Pensjonsavtaler({ pensjonsavtaler }: Props) {
-  if (pensjonsavtaler.length === 0) {
+export function Pensjonsavtaler({ pensjonsavtaler, showError }: Props) {
+  const navigate = useNavigate()
+  const harSamtykket = useAppSelector(selectSamtykke)
+
+  if (!showError && pensjonsavtaler.length === 0) {
     return null
   }
 
+  // PEK-97 TODO håndtere delvis reponse fra backend
   return (
     <Accordion.Item data-testid="pensjonsavtaler">
       <SectionHeader
         label="Pensjonsavtaler"
-        value={`${pensjonsavtaler.length}`}
+        value={getPensjonsavtalerTittel(
+          !!harSamtykket,
+          showError,
+          `${pensjonsavtaler.length}`
+        )}
```

Comment:
Innholdet i `SectionHeader` og `SectionContent` har 3 forskjellige varianter som avhenger av om bruker har samtykket eller om vi har en feil i henting av pensjonsavtaler. Det hadde blitt mye ryddigere å skille `Pensjonsavtaler` ut i 3 forskjellige komponenter som har ansvar for å å rendre henholdsvis melding om manglende samtykke, feilmelding ism. henting av pensjonavtaler, og pensjonsavtalene hvis alt er OK.

Reply:
Det er jo nettopp her at logikken er spredt. I både `SectionHeader` og `SectionContent` sjekker du status på samtykke og fetch og rendrer UI basert på dette. Da kan man heller ha én sjekk lenger oppe og så rendre 3 forskjellige komponenter basert på den ene sjekken i stedet for å ha samme sjekk to forskjellige steder i renderingkoden. Det blir mindre komplisert med én if-sjekk med tre forskjellige utfall vs. to if-sjekker spredt i jsx-koden som rendrer de samme tre forskjellige utfallene.

Reply:
At man har én ternary, og 3 ulike komponenter, eller at man har én komponent og 3 ternary i den er hipp som hopp. Det er ikke at jeg ikke vil ta imot den kommentaren, men jeg kan ikke kode det akkurat som du hadde gjort det selv, og det blir heller ikke mer riktig av den grunn. 

For ikke så lenge siden var det direkte push til main som var regelen, og nå har vi review som går på hvilket nivå man skal skrive sine ternaries? Jeg tror ikke at det er riktig at PR-review skal ligge på det nivået.

Reply:
Poenget mitt har hele tiden vært at det _ikke_ er hipp som happ, at koden blir mer kompleks når man har flere conditionals i jsx-koden sammenlignet med bare ett sted lenger opp. Man kan se på det matematisk: Én ternary lenger opp i koden = tre potensielle utfall vs. to ternaries i jsx-koden = ni (tilsynelatende) potensielle utfall. 

Med to ternaries er det ikke i realiteten her ni utfall siden det bare er samme sjekken begge steder, det er faktisk bare de samme tre utfallene som om man hadde hatt den ene sjekke lenger opp, men når man leser koden er ikke det umiddelbart synlig. Derfor er det forbedringspotensiale her når det kommer til lesbarhet og kompleksitet.

Om du velger å se bort i fra det her er det greit, skal ikke la det her blokkere PRen